### DISCUSSION TITLE
Getting Colors for Use In Code Behind or ViewModel

### DISCUSSION THREAD

**Discussion by seikosantana** [AvaloniaCommunity/Material.Avalonia#297] [Q&A] [answered]
I am working with a plotting library and I'm wondering if there's a way to get the color of the foreground text for current theme from viewmodel or code behind. I see that in XAML we could use Dynamic resource, but if i need to access the color manually?

Thanks in advance.

**Comment by SKProCH** [comment, marked as answer]
As far as i can remember, you can use `TryFindResource` or  `FindResource` inside your window/usercontrol/etc. If you need an observable use `GetResourceObservable`.

**Reply by seikosantana** [reply]
Thanks. Probably unrelated, is there a way to determine if BaseTheme is `Dark` or `Light` either in code behind or XAML?

**Reply by SKProCH** [reply]
Yes.
```
var materialTheme = Application.Current.LocateMaterialTheme<MaterialTheme>();
materialTheme.BaseTheme // or
materialTheme.ActuatBaseTheme
```

You can read more [here](https://github.com/AvaloniaCommunity/Material.Avalonia/wiki/Advanced-Theming).

**Reply by seikosantana** [reply]
Thanks a lot!
